datalad/support/handlerepo.py

```diff
@@ -288,4 +290,84 @@ def get_metadata(self, files=None):
+        raise NotImplementedError
+
+    def import_metadata(self, importer, files=None, data=None,
+                        about_uri=DLNS.this):
+        """Imports metadata to the datalad-conform metadata representation
+        of the handle.
+
+        Expects either `files` or `data` to be not `None`.
+
+        Parameters:
+        ___________
+        importer: class
+          the importer to be used; has to be a subclass of MetadataImporter
+        files: str or list of str
+          either a path to the file or directory to be imported or a list
+          containing paths to the files.
+        data: dict of list of str
+          a dictionary containing the metadata to be imported. The key is
+          expected to be the file name and the value its content as a list of
+          the file's lines as returned by `readlines()`.
+        about_uri: URIRef
+          uri of the entity the metadata is about. By default it's the handle
+          itself.
+        """
+
+        if not issubclass(importer, MetadataImporter):
+            raise TypeError("Not a MetadataImporter: " + str(importer))
+
+        # TODO: check whether cfg-file even exists, otherwise create a basic one.
+        cfg_graph = Graph().parse(opj(self.path, HANDLE_META_DIR,
+                                          REPO_CONFIG_FILE),
+                                      format="turtle")
+
+        # check for existing metadata sources to determine the name for the
+        # new one:
+        # TODO: the numbering is shit ;) Use a hash or sth.
+        src_name = "%s_import%d" % (self.name,
+                                    len([src for src in
+                                         cfg_graph.objects(about_uri,
+                                                               DLNS.usesSrc)])
+                                    + 1)
+
+        # graph containing just new config statements:
+        cfg_graph = Graph()
+
+        if files is not None and data is None:
+            # treat it as a metadata source, that can be used again later on.
+            src_node = URIRef(src_name)
+            # add config-entries for that source:
+            cfg_graph.add((about_uri, DLNS.usesSrc, src_node))
+            if isinstance(files, string_types):
+                cfg_graph.add((src_node, DLNS.usesFile, URIRef(files)))
+            elif isinstance(files, list):
+                [cfg_graph.add((src_node, DLNS.usesFile, URIRef(f)))
+                 for f in files]
+
+        elif files is None and data is not None:
+            # just metadata to read, nothing we can refer to later on
+            pass
+        else:
+            raise ValueError("Either 'files' or 'data' have to be passed.")
+
+        im = importer(target_class='Handle', about_class='Handle',
+                      about_uri=about_uri)
+        im.import_data(files=files, data=data)
+
+        # add new config statements:
+        im.get_graphs()[REPO_CONFIG_FILE[:-4]] += cfg_graph
```

Comment:
Yes, I just wanted to connect it to the file name constants, so there's nothing else to change in case we want to change names/paths. But should do it only once. You're right of course.